### PR TITLE
Us8448

### PR DESCRIPTION
**Pull request by farhadmak**

### CODE REVIEW

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 1]
```diff
@@ -6,17 +6,20 @@
           <h3>New Supporting Document</h3>
         </div>
         <div class="modal-body">
+          <div v-if='valid == false' class="error-message">
```
Comment (j-rewerts) [inline plugins/ideas/src/components/SupportingDocs.vue]: Nit: use double quotes

File: plugins/ideas/src/components/ViewInitiative.vue [hunk 2]
```diff
@@ -98,6 +135,24 @@ export default {
         console.error('Failed updating status description.')
         this.errors.push(err)
       })
+    },
+    updateSupportingDocType () {
+      var i
+      for (i in this.supportingDocs) {
+        if (this.supportingDocs[i].type === 1) {
+          this.supportingDocs[i].type = 'Business Cases'
+        }
```
Comment (j-rewerts) [inline plugins/ideas/src/components/ViewInitiative.vue]: Nit: add else if on ending bracket.

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 3]
```diff
@@ -6,17 +6,20 @@
...
+            Error: Check to make sure all fields were completed!
+          </div> 
           <label class="form-label">
             Title
-            <input id="title" class="form-control" v-model='form.title' placeholder='What would you like to name your supporting document?'>
+            <input id="title" class="form-control" required v-model='form.title' placeholder="What would you like to name your supporting document?">
```
Comment (j-rewerts) [inline plugins/ideas/src/components/SupportingDocs.vue]: Nit use double quotes

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 4]
```diff
@@ -6,17 +6,20 @@
...
           </label>
           <label class="form-label">
             URL
-            <input id="url" type='url' class="form-control" v-model='form.url' placeholder='Please enter the full url'>
+            <input id="url" type="url" class="form-control" required v-model='form.url' placeholder="Please enter the full url (i.e. http://)">
           </label>
           <label class="form-label">
             Type
-            <select class='form-control' v-model="form.type">
+            <select class='form-control' required v-model="form.type">
```
Comment (j-rewerts) [inline plugins/ideas/src/components/SupportingDocs.vue]: Nit

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 5]
```diff
@@ -41,30 +44,45 @@
...
+        this.sending = true
+        this.services.ideas.createSupportingDoc(
+          this.id,
+          this.form.title,
+          this.form.url,
+          this.form.type.replace(/\s/g, '')     // removes spaces to meet backend expectations
+        ).then(x => {
+          this.sending = false
+          this.$emit('close', this.form.title, this.form.url, this.form.type)
+        }).catch((err, y) => {
```
Comment (j-rewerts) [inline plugins/ideas/src/components/SupportingDocs.vue]: Remove the 'y' parameter.

File: plugins/ideas/src/components/ViewInitiative.vue [hunk 6]
```diff
@@ -71,14 +103,19 @@ export default {
...
       if (response && response[0]) {
         this.resources = response[0]
+        console.log(response[0])
       }
       if (response && response[1]) {
         this.steps = response[1]
       }
+      if (response && response[2]) {
+        this.supportingDocs = response[2]
+        this.updateSupportingDocType()
```
Comment (j-rewerts) [inline plugins/ideas/src/components/ViewInitiative.vue]: Remove

File: plugins/ideas/src/components/ViewInitiative.vue [hunk 7]
```diff
@@ -51,16 +69,30 @@ export default {
     resources: null,
     activeUser: null,
     canEditSteps: false,
-    supportingDocs: null,
-    showModal: false
+    supportingDocs: [],
+    showModal: false,
+    fakesupportingdoc: [{
```
Comment (j-rewerts) [inline plugins/ideas/src/components/ViewInitiative.vue]: Move mock data to StubbedIDeasService.

File: plugins/ideas/src/components/ViewInitiative.vue (hunk 2 above)
Comment (farhadmak) [reply plugins/ideas/src/components/ViewInitiative.vue]: done

File: plugins/ideas/src/components/SupportingDocs.vue (hunk 4 above)
Comment (farhadmak) [reply plugins/ideas/src/components/SupportingDocs.vue]: done

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 8]
```diff
@@ -41,30 +44,45 @@
...
+          this.sending = false
+          console.debug(err)
+          console.debug(y)
```
[the remark this answers is not in the document]
Comment (farhadmak) [reply plugins/ideas/src/components/SupportingDocs.vue]: done

File: plugins/ideas/src/components/SupportingDocs.vue [hunk 9]
```diff
@@ -25,7 +28,7 @@
           </label>
         </div>
         <div class="modal-footer text-right">
-          <md-button v-if='sending == false' v-on:click="$emit('close')" class='md-raised modal-default-button'>
+          <md-button v-if='sending == false' v-on:click="$emit('close')" class="md-raised modal-default-button">
```
[the remark this answers is not in the document]
Comment (farhadmak) [reply plugins/ideas/src/components/SupportingDocs.vue]: partially done. TODO fix font size

File: plugins/ideas/src/components/ViewInitiative.vue (hunk 7 above)
Comment (farhadmak) [reply plugins/ideas/src/components/ViewInitiative.vue]: done

**Pull request merged**